### DISCUSSION TITLE
Some additional features I've implemented for myself - is there interest in upstreaming them?

### DISCUSSION THREAD

**Discussion by nzig** [PatrickF1/fzf.fish#214] [Ideas]
First off, I really like this plugin and would like to thank @PatrickF1 for it 😊.

I've implemented a few features that I've wanted personally in my [fork](https://github.com/nzig/fzf.fish). The features I've added are:
- Add "magic mode" which tries to guess which kind of fzf to open (`Alt+M`). This is similar to the request in #103.
- Add an fzf mode to search for git branches (and tags)
- Allow multiple selection when selecting commits from git log (useful e.g. for `cherry-pick` or `revert`)
- When activating git log search, if the current token is a commit reference, show the git log at that commit instead of HEAD (otherwise use it as the initial fzf query as usual)
- Add an fzf mode to search for PIDs
- Add an fzf mode to search for SSH hostnames

This should not be considered stable code, but I will be using it personally.

I'd like to know if these would be useful to a more general audience. If so, please try them out in my fork and let me know what you think.

@PatrickF1 if you would like me to upstream any of these, I will split them out and create a PR.

**Comment by PatrickF1** [comment]
Hi @nzig ! You're welcome! I'm thankful to see someone appreciate my work and be so passionate about building on it!

Here are my thoughts about merging
- magic mode: pretty cool! my main issue is it's a lot of additional surface area to maintain. Unfortunately, I've reached a point where I am less ambitious and more "I just want to add solid new features slowly that everyone can use and like"
- git branches and tags: good idea, this is just lower priority for me because fish already has pretty good completion for branches
- multi select for git log: I should merge this in
- "When activating git log search, if the current token is a commit reference,": also a great idea but too complex/obscure for my tastes, but I bet @kidonng would love that for his own config as well
- PIDs: good idea, I am just stuck on figuring out how to make ps work on all platforms (seems every platform's ps is different)
- ssh hostnames: using `__fish_print_hostnames` is genius, I just need some time to better understand it, make sure it does what we want it to do, and that it'll be stable for a foundation for ssh hostnames

Thanks for all these ideas! I'm sorry if i come off as a wet blanket. Getting tons of bug reports and complaints about new features has made me jaded :/

**Reply by nzig** [reply]
No problem, thanks for the reply. Is there any part you would like me to submit as a PR? The multi select for git log?

**Reply by PatrickF1** [reply]
Sure I'd merge in a PR for multi select for git log.
And if you can propose a solution or provide some thoughts that assuage my concerns for PIDs and sshing hostnames, then I could also consider those features.

**Reply by PatrickF1** [reply]
Btw the pids is being worked on https://github.com/PatrickF1/fzf.fish/pull/216. Feel free to chime in with comments and feedback!
